src/payments-methods/express/payment-methods-express.js

```diff
@@ -14,19 +14,48 @@ import { renderAmazonButton } from '../../renderAmazonButton';
  * @returns React component
  */
 const AmazonPayExpressBtn = ( props ) => {
+	const estimatedOrderAmount = calculateEstimatedOrderAmount( props );
+
 	useEffect( () => {
-		renderAmazonButton( '#pay_with_amazon_express', 'express', null );
+		renderAmazonButton( '#pay_with_amazon_express', 'express', null, estimatedOrderAmount );
 	}, [] );
 
 	return <div id="pay_with_amazon_express" />;
 };
 
+/**
+ * Returns the estimated order amount button attribute.
+ * @param {object} props
+ * @returns {object}|null Estimated order amount button attribute.
+ */
+const calculateEstimatedOrderAmount = ( props ) => {
+	const { billing } = props;
+	const { currency } = billing;
+
+	const stringCartTotal     = String( billing.cartTotal.value );
+	const cartTotalLength     = stringCartTotal.length;
+	const decimals            = currency.minorUnit;
```

Comment:
Where `currency` is coming from?

Reply:
Found!

Reply:
Is being set 2 lines above on the same method. Originally, some `props` are being fed into the `calculateEstimatedOrderAmount` function. currency is located in `props.billing.currency`